src/mvhistory.jl

```diff
@@ -97,3 +99,16 @@ macro trace(tr, i, vars...)
     end
     block
 end
+
+"""
+    pushoradd!(trace, key, iter, val)
+
+Increments the value for a given key and iteration if it exists, otherwise adds the key/iteration pair with an ordinary push.
+"""
+function pushoradd!(trace::MultivalueHistory, key::Symbol, iter::Number, val)
+    if haskey(trace, key) && length(trace.storage[key]) >= iter
+        trace[key].values[iter] += val
```

Comment:
Accessing the member variable `values` is rather specific to `MultivalueHistory{<:History}` and would not work with `QHistory`. I am ok with that but we should catch this earlier I think. maybe dispatch the whole method on `trace::MultivalueHistory{<:History}`.

Reply:
I can't seem to figure out how to create an `MVHistory{QHistory}`?
Also, `trace::MultivalueHistory{<:History}` Gives me a type error, I can only get it to work with `trace::MVHistory{<:History}`